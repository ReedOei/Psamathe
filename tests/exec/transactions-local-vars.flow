contract Main {
    transaction foo() returns ! nat:
        var x : ! nat := 0
        47 --> x
        return 0
}

;

var x : empty Main
new Main() --> x
x.foo()

