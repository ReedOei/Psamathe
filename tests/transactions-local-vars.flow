contract Main {
    transaction foo() returns ! nat:
        var x : ! nat
        47 --> x
        return 0
}

;

var x : empty Main
new Main() --> x
x.foo()

