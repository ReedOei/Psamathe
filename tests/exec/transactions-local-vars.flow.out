<generatedTop>
  <common>
    <k>
      .
    </k>
    <contracts>
      Main |-> contract Main { transaction foo ( .VarDefs ) returns ! nat : var x : ! nat := 0 47 --[ everything ]-> x return 0  .Decls }
    </contracts>
    <fields>
      Main |-> .Set
    </fields>
    <constructors>
      Main |-> constructor ( .VarDefs , pass )
    </constructors>
    <functions>
      Main . foo |-> transaction ( var ( 0 ) : ! nat , .VarDefs , pass var x : ! nat := 0 in ( ( pass 47 --[ everything ]-> x ) pass 0 --[ everything ]-> var ( 0 ) pack exit ) )
    </functions>
    <types>
      address |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , address )
      bool |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , bool )
      nat |-> modified ( SetItem ( consumable )
      SetItem ( fungible ) , nat )
      string |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , string )
    </types>
    <freshCounter>
      7
    </freshCounter>
  </common>
  <lookup>
    this |-> loc ( -1 )
  </lookup>
  <storage>
    loc ( 2 ) |-> resource ( Main , .Map )
    loc ( 4 ) |-> resource ( nat , 47 )
    loc ( 5 ) |-> resource ( nat , 0 )
    loc ( 6 ) |-> resource ( nat , 0 )
    loc ( -1 ) |-> resource ( Base , .Map )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
